corehq/pillows/ledger.py

```diff
@@ -42,8 +42,6 @@ def _prepare_ledger_for_es(ledger):
 
     _update_ledger_section_entry_combinations(ledger)
 
-    return ledger
```

Comment:
This threw me at first - didn't expect `_prepare_ledger_for_es` to do a stateful operation.  I see now that `_update_ledger_section_entry_combinations` does, though.

Reply:
Not sure if you noticed but `_update_ledger_section_entry_combinations` was not the only stateful operation here. `_get_daily_consumption_for_ledger` also updates the daily consumption for the ledger https://github.com/dimagi/commcare-hq/blob/f453bc326a5f9571679bceeff2f9455ce1efee64/corehq/pillows/ledger.py#L57

This was also a surprise to me and I only found out about when I was removing tests in my original PR. For example: https://github.com/dimagi/commcare-hq/blob/f453bc326a5f9571679bceeff2f9455ce1efee64/corehq/apps/commtrack/tests/test_xml.py#L239